core/services/ocr2/plugins/ocr2keeper/evm21/registry_check_pipeline.go

```diff
@@ -101,7 +101,7 @@ func (r *EvmRegistry) getBlockHash(blockNumber *big.Int) (common.Hash, error) {
 }
 
 // verifyCheckBlock checks that the check block and hash are valid, returns the pipeline execution state and retryable
-func (r *EvmRegistry) verifyCheckBlock(_ context.Context, checkBlock, upkeepId *big.Int, checkHash common.Hash) (state encoding.PipelineExecutionState, retryable bool) {
+func (r *EvmRegistry) verifyCheckBlock(_ context.Context, checkBlock, upkeepId *big.Int, checkHash common.Hash) (state uint8, retryable bool) {
```

Comment:
Here and many other places, a transition away from defined types to base types was done. Was there a reason for this? We lose some value of strong types in static code analysis when we do this.

Reply:
It is also because of circular dependencies. The types are defined in `encoding` package, and we dont want to reference them from mercury packages. The `UpkeepFailureReason` is used by both mercury and non-mercury. Happy to hear other approaches. 

```
type UpkeepFailureReason uint8
type PipelineExecutionState uint8
```

Reply:
As part of the spike, it looked like the defined types were always explicitly cast to their underlying types before use, and we didn't seem to define any custom functions on the new types so it didn't feel necessary to keep the defined types at the time, but it's not something I feel strongly about either